### PR TITLE
Conditionally load Resource Hints for font domains

### LINKED ISSUE #3851: Conditionally load Resource Hints for font domains
**Is your feature request related to a problem? Please describe.**
We currently have resource hints <link rel="preconnect" > & <link rel="dns-preconnect" > for several assets, including font domains. 
We have these for all pages, regardless of whether they have fonts.

We should only conditionally load these.

**Describe the solution you'd like**
One possibility: pass in the service config's fonts object and include the relevant font domains. 

**Describe alternatives you've considered**
We could leave it as it is currently. 

**Testing notes**
[Tester to complete]

Dev insight: Unit and integration tests should be sufficient

- [ ] This feature is expected to need manual testing.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by EinsteinNjoroge**
Resolves #3851 

**Overall change:** _Load Resource Hints for font domains only on services where `fonts` have been defined in service config._

**Code changes:**

- _Move `serviceConfig` from `testHeplers` to `server/utilities` so that we can use in in production code._
- _Import serviceConfigs into `getAssetOrigins` utility._
- _Load fonts origins if fonts have been defined in service config._
---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by hotinglok — COMMENTED** [review]
Still making my way through this but just wanted to note that it might be worth making a webpack alias for `utilities` as well.

**Review by RayNjeri — APPROVED** [review]
Well done :-D

File: cypress/integration/application/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import config from '../../support/config/services';
-import appConfig from '../../../src/testHelpers/serviceConfigs';
+import appConfig from '../../../src/server/utilities/serviceConfigs';
```
Comment (hotinglok) [inline cypress/integration/application/index.js:2]: Can't this line also make use of the webpack alias you made? Or does cypress not like that? 
LGTM besides this, nice work!

Comment (EinsteinNjoroge) [reply cypress/integration/application/index.js:2]: Cypress runs on a different set of configurations, using aliases here will lead to unresolved imports.

🤔 Unless we configure Cypress to use our webpack configuration. I'm not sure if that's a good idea.

Comment (hotinglok) [reply cypress/integration/application/index.js:2]: Ah I see, I didn't know that, thanks! I'll approve it now

**Review by hotinglok — APPROVED** [review]

File: cypress/integration/pages/testsForAllPages.js [hunk 2]
```diff
@@ -19,12 +19,15 @@ export const testsThatFollowSmokeTestConfigforAllPages = ({
...
       it('should have resource hints', () => {
-        const resources = [
-          envConfig.assetOrigin,
-          'https://ichef.bbci.co.uk',
-          'https://gel.files.bbci.co.uk',
-          'https://ws-downloads.files.bbci.co.uk',
-        ];
+        const resources = [envConfig.assetOrigin, 'https://ichef.bbci.co.uk'];
+        const serviceConfig = appConfig[service];
+        const { fonts } = serviceConfig[variant || 'default'];
```
Comment (jroebu14) [inline cypress/integration/pages/testsForAllPages.js:24]: I think the default value for `variant` should be set in the function signature.

**Review by jroebu14 — APPROVED** [review]

File: dirAlias.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ module.exports = {
     '#lib': resolvePath('src/app/lib/'),
     '#models': resolvePath('src/app/models/'),
     '#testHelpers': resolvePath('src/testHelpers/'),
+    '#utilities': resolvePath('src/server/utilities/'),
```
Comment (dr3) [inline dirAlias.js]: This should be clear its server utilities, if someone imports `#utilities/serviceConfigs` it will break the client app

**Review by dr3 — CHANGES_REQUESTED** [review]

File: dirAlias.js (hunk 3 above)
Comment (dr3) [reply dirAlias.js]: I suggest we go with `#server` and use `#server/utilities`

Comment (EinsteinNjoroge) [reply dirAlias.js]: Makes sense

**Review by dr3 — APPROVED** [review]
Approving for the requested changes I asked for :)

**Comment by EinsteinNjoroge** [comment]
Can't merge this PR due to unsigned commits, I have reopened this PR with signed commits [here](https://github.com/bbc/simorgh/pull/4365)

**Pull request closed without merging**